src/middleware.js

```diff
@@ -31,55 +39,42 @@ export default function wrapper(context) {
       });
     }
 
-    const acceptedMethods = context.options.methods || ['GET', 'HEAD'];
-
-    if (acceptedMethods.indexOf(req.method) === -1) {
-      return goNext();
-    }
-
-    return new Promise((resolve) => {
-      // eslint-disable-next-line consistent-return
-      function processRequest() {
-        const filename = getFilenameFromUrl(context, req.url);
-
-        if (!filename) {
-          return resolve(goNext());
-        }
+    async function processRequest() {
+      const filename = getFilenameFromUrl(context, req.url);
+      const { headers } = context.options;
+      let content;
 
-        let content;
+      if (!filename) {
+        await goNext();
+        return;
+      }
 
-        try {
-          content = context.outputFileSystem.readFileSync(filename);
-        } catch (_ignoreError) {
-          return resolve(goNext());
-        }
+      try {
+        content = context.outputFileSystem.readFileSync(filename);
+      } catch (_ignoreError) {
+        await goNext();
+        return;
+      }
 
-        content = handleRangeHeaders(content, req, res);
+      content = handleRangeHeaders(content, req, res);
 
-        if (!res.get('Content-Type')) {
-          const contentType = mime.contentType(path.extname(filename));
+      if (!res.get('Content-Type')) {
+        const contentType = mime.contentType(path.extname(filename));
 
-          if (contentType) {
-            res.set('Content-Type', contentType);
-          }
+        if (contentType) {
+          res.set('Content-Type', contentType);
         }
+      }
 
-        const { headers } = context.options;
-
-        if (headers) {
-          for (const name in headers) {
-            if ({}.hasOwnProperty.call(headers, name)) {
-              res.set(name, context.options.headers[name]);
-            }
+      if (headers) {
+        for (const name in headers) {
+          if (Reflect.has(headers, name)) {
```

Comment:
Bad idea

Reply:
Why?

Reply:
`{}.hasOwnProperty.call(headers, name)` is very poorly readable.

Reply:
Hm, looks `Reflect` has good supporting, so it is good to use it

Reply:
Actually, this code is unnecessary. I updated.